cpp/src/parquet/column_writer_test.cc

```diff
@@ -1705,5 +1719,55 @@ TEST(TestColumnWriter, WriteDataPageV2HeaderNullCount) {
   }
 }
 
+using TestInt32Writer = TestPrimitiveWriter<Int32Type>;
+
+TEST_F(TestInt32Writer, NoWriteKeyValueMetadata) {
+  auto writer = this->BuildWriter();
+  writer->Close();
+  auto key_value_metadata = metadata_key_value_metadata();
+  ASSERT_THAT(key_value_metadata, IsNull());
+}
+
+TEST_F(TestInt32Writer, WriteKeyValueMetadata) {
+  auto writer = this->BuildWriter();
+  writer->AddKeyValueMetadata(KeyValueMetadata::Make({"hello"}, {"world"}));
+  writer->Close();
+  auto key_value_metadata = metadata_key_value_metadata();
+  ASSERT_THAT(key_value_metadata, NotNull());
+  ASSERT_THAT(key_value_metadata->size(), 1);
+  ASSERT_OK_AND_ASSIGN(auto value, key_value_metadata->Get("hello"));
+  ASSERT_THAT(value, "world");
```

Comment:
Should we use `ASSERT_EQ` here so that the test is more readable? Currently it's a bit confusing what's being checked exactly.

Reply:
I'm against using ASSERT_EQ because the direction is reversed, e.g., `ASSERT_EQ(expected, actual)`, `ASSERT_THAT(actual, matcher)`, and it can hurt the readability when you have both ASSERT_EQ and ASSERT_THAT in the same code. If the readability is a concern, we can write `ASSERT_THAT(actual, Eq(expected))`. What do you thin?

Reply:
I'm ok with `ASSERT_THAT(actual, Eq(expected))`